source/magpie/dataframe.d

```diff
@@ -188,51 +214,41 @@ public:
                     maxGap = (maxGap > lenCol)? maxGap: lenCol;
                 }
 
+                size_t maxsize1 = 0, maxsize2 = 0;
                 static if(isHomogeneousType)
                 {
-                    size_t maxsize1 = 0, maxsize2 = 0;
                     if(top > indx.column.index.length + extra)
                         maxsize1 = data[dataIndex][0 .. top - indx.column.index.length - extra].map!(e => to!string(e).length).reduce!max;
                     if(bottom > data[dataIndex].length)
                         maxsize2 = data[dataIndex].map!(e => to!string(e).length).reduce!max;
                     else if(bottom > 0)
-                        maxsize2 = data[dataIndex][$ - bottom .. $].map!(e => to!string(e).length).reduce!max;
-
-                    if(maxsize1 > maxGap)
-                    {
-                        maxGap = maxsize1;
-                    }
-                    if(maxsize2 > maxGap)
-                    {
-                        maxGap = maxsize2;
-                    }
+                        maxsize2 = data[dataIndex][$ - bottom .. $].map!(e => to!string(e).length).reduce!max;   
```

Comment:
can I suggest making `T[]..map!(e => to!string(e).length).reduce!max` a function and then calling it on the computed ranges.
e.g.

```
/*use a better name*/
auto transform(T)(T[] arr){ return arr.map!(e => to!string(e).length).reduce!max;}
...
if(bottom > data[dataIndex].length)
    maxsize2 = transform(data[dataIndex]);
else if (bottom > 0) 
    maxsize2 = transform(data[dataIndex][$ - bottom .. $]);
```

also computing `to!string(e)` just for the length seems wasteful if you don't use the result. This is totally fine for getting stuff up and running quickly, but you should do some profiling to determine the bottlenecks of this library. 

I suspect GC allocation may be one of them, but do the profiling anyway because it i entirely possible I'm wrong and that the optimiser inlines everything and discards the useless allocation (use LDC for that, DMD won't do much to help with that).